crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -51,6 +56,209 @@ use sui_types::sui_serde::{Base64, Encoding};
 mod rpc_types_tests;
 
 pub type GatewayTxSeqNumber = u64;
+pub type SuiMoveTypeParameterIndex = u16;
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveAbilitySet(u8);
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub enum SuiMoveVisibility {
+    Private = 0x0,
+    Public = 0x1,
+    Friend = 0x3,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveStructTypeParameter {
+    pub constraints: SuiMoveAbilitySet,
+    pub is_phantom: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedField {
+    pub name: String,
+    pub type_: MoveNormalizedType,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedStruct {
+    pub abilities: SuiMoveAbilitySet,
+    pub type_parameters: Vec<SuiMoveStructTypeParameter>,
+    pub fields: Vec<MoveNormalizedField>,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub enum MoveNormalizedType {
+    Bool,
+    U8,
+    U64,
+    U128,
+    Address,
+    Signer,
+    Struct {
+        address: String,
+        module: String,
+        name: String,
+        type_arguments: Vec<MoveNormalizedType>,
+    },
+    Vector(Box<MoveNormalizedType>),
+    TypeParameter(SuiMoveTypeParameterIndex),
+    Reference(Box<MoveNormalizedType>),
+    MutableReference(Box<MoveNormalizedType>),
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedFunction {
+    pub visibility: SuiMoveVisibility,
+    pub is_entry: bool,
+    pub type_parameters: Vec<SuiMoveAbilitySet>,
+    pub parameters: Vec<MoveNormalizedType>,
+    pub return_: Vec<MoveNormalizedType>,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveModuleId {
+    address: String,
+    name: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedModule {
+    pub file_format_version: u32,
+    pub address: String,
+    pub name: String,
+    pub friends: Vec<SuiMoveModuleId>,
+    pub structs: BTreeMap<String, MoveNormalizedStruct>,
+    pub exposed_functions: BTreeMap<String, MoveNormalizedFunction>,
```

Comment:
Correct me if I am wrong, it looks like these only include entry functions. Is it possible to get the info for non-entry functions? cc @todd-mystenlabs @emmazzz @sblackshear

Reply:
from test above, the functions in 0x2 coin module is something like
```
"exposed_functions":{

"balance":{"visibility":"Public","is_entry":false,"type_parameters":[0],"parameters":[{"Reference":{"Struct":{"address":"0x2","module":"coin","name":"Coin","type_arguments":[{"TypeParameter":0}]}}}],"return_":[{"Reference":{"Struct":{"address":"0x2","module":"balance","name":"Balance","type_arguments":[{"TypeParameter":0}]}}}]},
"balance_mut":{"visibility":"Public","is_entry":false,"type_parameters":[0],"parameters":[{"MutableReference":{"Struct":{"address":"0x2","module":"coin","name":"Coin","type_arguments":[{"TypeParameter":0}]}}}],"return_":[{"MutableReference":{"Struct":{"address":"0x2","module":"balance","name":"Balance","type_arguments":[{"TypeParameter":0}]}}}]},
"burn":{"visibility":"Public","is_entry":false,"type_parameters":[0],"parameters":[{"MutableReference":{"Struct":{"address":"0x2","module":"coin","name":"TreasuryCap","type_arguments":[{"TypeParameter":0}]}}},{"Struct":{"address":"0x2","module":"coin","name":"Coin","type_arguments":[{"TypeParameter":0}]}}],"return_":["U64"]},
"burn_":{"visibility":"Public","is_entry":true,"type_parameters":[0],"parameters":[{"MutableReference":{"Struct":{"address":"0x2","module":"coin","name":"TreasuryCap","type_arguments":[{"TypeParameter":0}]}}},{"Struct":{"address":"0x2","module":"coin","name":"Coin","type_arguments":[{"TypeParameter":0}]}}],"return_":[]},
...

}
```

Reply:
Got it. Previously I tested with this contract https://github.com/666lcz/toy-move-packages/blob/main/fungible-tokens/sources/usdc.move and the `init` function was not included in the `NormalizedModule` content. Is it expected that `init` is not included? cc @todd-mystenlabs

Reply:
by default, I believe private functions are not included in the NormalizedModule

You probably want to include them and do some filtering, since `entry` functions might be private

Reply:
@666lcz do we need to show the entry function even if it is private? 
if we want to include them, we will have to 1. change the normalized struct 2. mimic the normalization on Move side in Sui to pick the entry function and merge it into `exposed_function`. This can be a bit tricky.

Reply:
Yes, ideally we want to show the type info for private functions, given that `init` functions are usually at the top of the file and it feels a bit broken if their arguments are not clickable. @tnowacki could you provide some guidance on how to get this info?

Reply:
> @666lcz do we need to show the entry function even if it is private?

You should as they can still serve as entry functions for transactions. I think it is debatable for non-entry private functions.

Reply:
Looks like `entry` functions are included, but yeah, we will need a PR to allow for custom filtering (like including private functions)